managers/bosh-manager/DirectorService.js

```diff
@@ -0,0 +1,1112 @@
+        Setting default platform as '${CONST.PLATFORM.CF}'`);
+
+        const context = {
+          platform: CONST.PLATFORM.CF
+        };
+        return context;
+      });
+  }
+
+  static get prefix() {
+    return _
+      .reduce(config.directors,
+        (prefix, director) => director.primary === true ? director.prefix : prefix,
+        null) || super.prefix;
+  }
+
+  get deploymentName() {
+    return this.getDirectorDeploymentName(this.guid, this.networkSegmentIndex);
+  }
+
+  getDirectorDeploymentName(guid, networkSegmentIndex) {
+    let subnet = this.subnet ? `_${this.subnet}` : '';
+    return `${this.prefix}${subnet}-${NetworkSegmentIndex.adjust(networkSegmentIndex)}-${guid}`;
+  }
+
+  getNetworkSegmentIndex(deploymentName) {
+    return _.nth(BaseDirectorService.parseDeploymentName(deploymentName, this.subnet), 1);
+  }
+
+  getInstanceGuid(deploymentName) {
+    return _.nth(BaseDirectorService.parseDeploymentName(deploymentName, this.subnet), 2);
+  }
+
+  initialize(operation) {
+    return Promise
+      .try(() => {
+        this.operation = operation.type;
+        if (operation.type === CONST.OPERATION_TYPE.CREATE) {
+          return this.acquireNetworkSegmentIndex(this.guid);
+        }
+        return this.findNetworkSegmentIndex(this.guid);
+      })
+      .tap(networkSegmentIndex => {
+        assert.ok(_.isInteger(networkSegmentIndex), `Network segment index '${networkSegmentIndex}' must be an integer`);
+        this.networkSegmentIndex = networkSegmentIndex;
+      })
+      .tap(() => {
+        if (operation.type === 'delete') {
+          return Promise
+            .all([
+              this.platformManager.preInstanceDeleteOperations({
+                guid: this.guid
+              }),
+              this.deleteRestoreFile()
+            ]);
+        }
+      });
+  }
+
+  acquireNetworkSegmentIndex(guid) {
+    logger.info(`Acquiring network segment index for a new deployment with instance id '${guid}'...`);
+    const promises = [this.getDeploymentNames(true)];
+    if (config.enable_bosh_rate_limit) {
+      promises.push(this.getDeploymentNamesInCache());
+    }
+    return Promise.all(promises)
+      .then(deploymentNameCollection => _.flatten(deploymentNameCollection))
+      .then(deploymentNames => {
+        const deploymentName = _.find(deploymentNames, name => _.endsWith(name, guid));
+        if (deploymentName) {
+          logger.warn('+-> Deployment with this instance id already exists');
+          throw new ServiceInstanceAlreadyExists(guid);
+        }
+        return NetworkSegmentIndex.findFreeIndex(deploymentNames, this.subnet);
+      })
+      .tap(networkSegmentIndex => logger.info(`+-> Acquired network segment index '${networkSegmentIndex}'`));
+  }
+
+  deleteRestoreFile() {
+    if (_.includes(this.agent.features, 'backup')) {
+      return Promise.try(() => this.platformManager.ensureTenantId({
```

Comment:
usage of platform manager.

Reply:
Platform managers would still remain. SF2.0 managers are for the provisioning use case. Platform managers are for the consuming platform specific use case like SG creation. Cant be removed.